flaskbb/forum/views.py

```diff
@@ -143,6 +143,16 @@ def get(self, topic_id, slug=None):
         topic.views += 1
         topic.save()
 
+
+        # Update the topicsread status if the user hasn't read it
+        forumsread = None
```

Comment:
Moved this up since `Topic.update_read` can commit which flushes the cache which means we might need to go to the database again for the Posts.